### ISSUE TITLE
[6.1.1] Bug: ARRAY_* reducers incorrectly uses splice on `state.fields` objects

### ISSUE THREAD

**Issue by huan086** [erikras/redux-form#2014]
redux-form state.fields properties are objects. Fields for arrays are stored as { 0: { prop1: { touched: true }}, touched: true}. However, in reducer.js, ARRAY_\* operates on this object as if it were an array [result = doSplice(result, 'fields', field, index, removeNum, nonValuesValue)](https://github.com/erikras/redux-form/blob/master/src/reducer.js#L59).

state.fields is thus corrupted after that.

**Comment by huan086** [comment]
# Proposed solution

during REGISTER_FIELD, if it is FieldArray registering, we set state.fields[name] to []

**Comment by huan086** [comment]
This seems to solve the problem. Insert at line 225

```jsx
            if (type === "FieldArray") {
                const fields = getIn(`fields.${name}`);
                if (!fields) {
                    result = setIn(result, `fields.${name}`, []);
                }
            }
```

**Comment by clayne11** [comment]
If you have a solution please create a PR with a test showing the broken issue and how your change fixes it. PRs are merged in pretty frequently.

**Comment by erikras** [comment]
Fix released as [`v6.5.0`](https://github.com/erikras/redux-form/releases/tag/v6.5.0).
